modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/robotic/synthetic.dm

```diff
@@ -0,0 +1,161 @@
+	. = ..()
+	var/obj/item/organ/internal/appendix/appendix = transformer.getorganslot(ORGAN_SLOT_APPENDIX)
+	if(appendix)
+		appendix.Remove(transformer)
+		qdel(appendix)
+
+	var/screen_mutant_bodypart = transformer.dna.mutant_bodyparts[MUTANT_SYNTH_SCREEN]
+
+	if(!screen && screen_mutant_bodypart && screen_mutant_bodypart[MUTANT_INDEX_NAME] && screen_mutant_bodypart[MUTANT_INDEX_NAME] != "None")
+		screen = new
+		screen.Grant(transformer)
+
+/datum/species/synthetic/replace_body(mob/living/carbon/target, datum/species/new_species)
+	. = ..()
+	var/list/chassis = target.dna.mutant_bodyparts[MUTANT_SYNTH_CHASSIS]
+	var/list/head = target.dna.mutant_bodyparts[MUTANT_SYNTH_HEAD]
+	if(!chassis && !head)
+		return
+
+	var/datum/sprite_accessory/synth_chassis/chassis_of_choice = GLOB.sprite_accessories[MUTANT_SYNTH_CHASSIS][chassis[MUTANT_INDEX_NAME]]
+	var/datum/sprite_accessory/synth_head/head_of_choice = GLOB.sprite_accessories[MUTANT_SYNTH_HEAD][head[MUTANT_INDEX_NAME]]
+	if(!chassis_of_choice && !head_of_choice)
+		return
+
+	examine_limb_id = chassis_of_choice.icon_state
+
+	if(chassis_of_choice.color_src || head_of_choice.color_src)
+		species_traits += MUTCOLORS
+
+	// We want to ensure that the IPC gets their chassis and their head correctly.
+	for(var/obj/item/bodypart/limb as anything in target.bodyparts)
+		if(initial(limb.limb_id) != SPECIES_SYNTH && initial(limb.base_limb_id) != SPECIES_SYNTH) // No messing with limbs that aren't actually synthetic.
+			continue
+
+		if(limb.body_zone == BODY_ZONE_HEAD)
+			if(head_of_choice.color_src && head[MUTANT_INDEX_COLOR_LIST] && length(head[MUTANT_INDEX_COLOR_LIST]))
+				limb.variable_color = head[MUTANT_INDEX_COLOR_LIST][1]
+			limb.change_appearance(head_of_choice.icon, head_of_choice.icon_state, !!head_of_choice.color_src, head_of_choice.dimorphic)
+			continue
+
+		if(chassis_of_choice.color_src && chassis[MUTANT_INDEX_COLOR_LIST] && length(chassis[MUTANT_INDEX_COLOR_LIST]))
+			limb.variable_color = chassis[MUTANT_INDEX_COLOR_LIST][1]
+		limb.change_appearance(chassis_of_choice.icon, chassis_of_choice.icon_state, !!chassis_of_choice.color_src, limb.body_part == CHEST && chassis_of_choice.dimorphic)
+		limb.name = "\improper[chassis_of_choice.name] [parse_zone(limb.body_zone)]"
+
+/datum/species/synthetic/on_species_loss(mob/living/carbon/human/human)
+	. = ..()
+	if(screen)
+		screen.Remove(human)
+
+/**
+ * Simple proc to switch the screen of a monitor-enabled synth, while updating their appearance.
+ *
+ * Arguments:
+ * * transformer - The human that will be affected by the screen change (read: IPC).
+ * * screen_name - The name of the screen to switch the ipc_screen mutant bodypart to.
+ */
+/datum/species/synthetic/proc/switch_to_screen(mob/living/carbon/human/tranformer, screen_name)
+	tranformer.dna.mutant_bodyparts[MUTANT_SYNTH_SCREEN][MUTANT_INDEX_NAME] = screen_name
+	if(screen)
+		tranformer.update_body()
```

Comment:
How does that work for those that don't have the screen enabled?

Reply:
Any screen updates do absolutely nothing on mobs without screens, it updates the state in their DNA just in case they somehow gain a screen, I suppose. Though you do raise a good point, and I'll move that DNA setter into the check to prevent excess cycles being used on something that'll have no impact.